regcomp.h

```diff
@@ -1559,12 +1559,12 @@ typedef enum {
  * and this is not a DEBUGGING enabled build (identified by
  * DEBUGGING_RE_ONLY being defined)
  */
-#if ( defined(USE_DYNAMIC_LOADING) && defined(DEBUGGING)) || \
+#if ( !defined(PERL_EXT_RE_STATIC) && defined(DEBUGGING)) ||       \
     ( defined(PERL_EXT_RE_BUILD) && defined(DEBUGGING_RE_ONLY)) || \
     (!defined(PERL_EXT_RE_BUILD) && defined(DEBUGGING))
 #define PERL_RE_BUILD_DEBUG
 #endif
-#if ( defined(USE_DYNAMIC_LOADING) || !defined(PERL_EXT_RE_BUILD) )
```

Comment:
If this weren't you @tonycoz I would have thought this was wrong. You sure about this one?

Reply:
I'm not sure which part you think is wrong, so I'll go through it step by step, most of this you probably already know:

First, defining `PERL_EXT_RE_STATIC`:

1. when building with -Uusedl, all dynamic extensions including re are added to `static_ext`, so whether you do `-Dstatic_ext=re` or `-Uusedl`, `re` will be in `static_ext`.
2. for `static_ext` extensions Makefile.SH generates a rule that calls `make_ext.pl` with `LINKTYPE=static` to build the extension
3. my change makes `ext/re/Makefile.PL` define `PERL_EXT_RE_STATIC` if `Makefile.PL` sees `LINKTYPE=static`

As the `.h` change, we have four cases from the original condition:

| USE_DYNAMIC_LOADING | PERL_EXT_RE_BUILD | PERL_EXT_RE_STATIC | Want PERL_RE_BUILD_AUX
|--------|--------|--------|--------|
| Y | Y | N | Y |
| Y | N | N | Y |
| N | Y | Y | N |
| N | N | N | Y |

Based on the old condition we want PERL_RE_BUILD_AUX when we're dynamically loading re, or not building re itself.

PERL_EXT_RE_STATIC is only defined when re is being built and that build is static, so ignoring the explicit `static_ext=re` case:
```
PERL_EXT_RE_STATIC = !USE_DYNAMIC_LOADING &&PERL_EXT_RE_BUILD)
                   # deMorgan's Law
                   = !(USE_DYNAMIC_LOADING || !PERL_EXT_RE_BUILD)
```
so the inner part matches the original condition, and the change negates the condition so it matches.

Since the definition of PERL_EXT_RE_STATIC takes `-Dstatic_ext=re` into account, both the old case you fixed (`-Uusedl`) and the left over `-Dstatic_ext=re` case are covered.

Please let me know if I've made a mistake here.

Tony
(who forgot deMorgan's name)